### ISSUE TITLE
[Bug]: [Nightly]Search binary with 'tanimoto' metric type lost data occasionally 

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#22403]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.2 (7808bb5)
- Deployment mode(standalone or cluster):both
- MQ type(rocksmq, pulsar or kafka):    rocksmq, kafka
- SDK version(e.g. pymilvus v2.0.0rc2):2.2.3.dev1
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Search binary with 'tanimoto' metric type lost data occasionally 
```
[2023-02-23T21:29:58.183Z] [2023-02-23 20:35:06 - DEBUG - ci_test]: (api_request)  : [Collection.search] args: [[b'\x06\xf6l\xaa\x07u\xa8\x18ji\x13\x8f\xbb]$w', b'c\xf0\x8f\xe2\xbe\xbc\x97\xff@]~\x16\x13\x08\xe7E'], 'binary_vector', {'metric_type': 'TANIMOTO', 'params': {'nprobe': 10}}, 10, 'int64 >= 0', None, None, 120, -1], kwargs: {} (api_request.py:56)

[2023-02-23T21:29:58.183Z] [2023-02-23 20:35:06 - DEBUG - ci_test]: (api_response) : ["['(distance: 0.983880341053009, id: 780)', '(distance: 1.0641303062438965, id: 559)', '(distance: 1.0834159851074219, id: 384)', '(distance: 1.093109369277954, id: 2116)', '(distance: 1.095157265663147, id: 89)', '(distance: 1.1312445402145386, id: 957)', '(distance: 1.1312445402145386, id: 1660)'......  (api_request.py:31)

[2023-02-23T21:29:58.183Z] [2023-02-23 20:35:06 - INFO - ci_test]: search_results_check: checking the searching results (func_check.py:236)

[2023-02-23T21:29:58.183Z] [2023-02-23 20:35:06 - INFO - ci_test]: search_results_check: Numbers of query searched is correct (func_check.py:251)

[2023-02-23T21:29:58.183Z] [2023-02-23 20:35:06 - ERROR - ci_test]: search_results_check: limit(topK) searched (0) is not equal with expected (10) (func_check.py:255)
```

### Expected Behavior

Search successfully

### Steps To Reproduce

```markdown
testcases/test_search.py::TestCollectionSearch::test_search_binary_without_flush[TANIMOTO-False]:


   @pytest.mark.tags(CaseLabel.L2)
    def test_search_binary_without_flush(self, metrics, auto_id):
        """
        target: test search without flush for binary data (no index)
        method: create connection, collection, insert, load and search
        expected: search successfully with limit(topK)
        """
        # 1. initialize a collection without data
        collection_w = self.init_collection_general(prefix, is_binary=True, auto_id=auto_id)[0]
        # 2. insert data
        insert_ids = cf.insert_data(collection_w, default_nb, is_binary=True, auto_id=auto_id)[3]
        # 3. load data
        collection_w.load()
        # 4. search
        log.info("test_search_binary_without_flush: searching collection %s" % collection_w.name)
        binary_vectors = cf.gen_binary_vectors(default_nq, default_dim)[1]
        search_params = {"metric_type": metrics, "params": {"nprobe": 10}}
        collection_w.search(binary_vectors[:default_nq], "binary_vector",
                            search_params, default_limit,
                            default_search_exp,
                            check_task=CheckTasks.check_search_results,
                            check_items={"nq": default_nq,
                                         "ids": insert_ids,
                                         "limit": default_limit})
```

### Milvus Log

[artifacts-milvus-standalone-nightly-105-pymilvus-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20Nightly%20CI/job/2.2.0/105/artifact/artifacts-milvus-standalone-nightly-105-pymilvus-e2e-logs.tar.gz)

### Anything else?

1. collection name: search_collection_KC4mjOZI
2. failed time: 
[2023-02-23T20:35:06.548Z] [gw1] [ 60%] FAILED testcases/test_search.py::TestCollectionSearch::test_search_binary_without_flush[TANIMOTO-False]

**Comment by xiaofan-luan** [comment]
@cydrain pls make sure this is also fixed in knowhere 2.0

**Comment by NicoYuan1986** [comment]
Fixed.
